app/src/app.py

```diff
@@ -1,7 +1,51 @@
-from flask import Flask, render_template
-
+from flask import Flask, render_template, request, redirect, flash, url_for
+from werkzeug.security import generate_password_hash
 app = Flask(__name__)
 
 @app.route('/')
 def index():
     return render_template('index.html')
+
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    """Register user"""
+
+    if request.method == "POST":
+
+        #Access the data submitted to form
+        email = request.form.get("email")
+        username = request.form.get("username")
+        password = request.form.get("password")
+
+
+        #Check if the inputs are valid
+        user_email = User.query.filter_by(email=email).first()
+        if user_email:
+            flash("This email already exists.")
+            return redirect(url_for("register"))
+        user_name = User.query.filter_by(username=username).first()
+        if user_name:
+            flash("This username already exists.")
+            return redirect(url_for("register"))
+        if len(username) == 0:
+            flash("You must provide a username.")
+            return redirect(url_for("register"))
+        if len(password) == 0:
+            flash("You must provide a password.")
+            return redirect(url_for("register"))
+
+
+        #Add the user to database and login
+        new_user = User(email=email, username=username, password=generate_password_hash(password))
+        db.session.add(new_user)
+        db.session.commit()
+        flash("Registration Successful!")
```

Comment:
It could confuse a user to be redirected to the login page after registering so for now, I think, have a placeholder Wychwood redirect to the main page or the user profile page

Reply:
This is for line 44